components/specialized-parser/src/runner.py

```diff
@@ -0,0 +1,412 @@
+import csv
+import json
+import os
+import re
+import uuid
+from collections import namedtuple
+from dataclasses import asdict, dataclass
+from typing import Dict, List, Tuple
+
+import pg8000
+import sqlalchemy
+from configs import AlloyDBConfig, BigQueryConfig, JobConfig, ProcessorConfig
+from google.api_core.client_info import ClientInfo
+from google.api_core.client_options import ClientOptions
+from google.api_core.exceptions import GoogleAPICallError, RetryError
+from google.api_core.gapic_v1.client_info import ClientInfo
+from google.api_core.operation import Operation
+from google.cloud import bigquery, documentai, storage
+from google.cloud.alloydb.connector import Connector, IPTypes
+from google.cloud.documentai_v1 import BatchProcessMetadata
+from google.cloud.exceptions import InternalServerError
+from sqlalchemy.engine import Engine
+
+FilenamesPair = namedtuple("FilenamesPair", "original_filename txt_filename")
+
+
+PROCESSED_DOCUMENTS_TABLE_NAME = "eks.processed_documents"
+
+
+@dataclass
+class ProcessedDocument:
+    id: str
+    original_filename: str
+    results_file: str
+    run_id: str
+    entities: str
+
+
+USER_AGENT = "cloud-solutions/eks-docai-v1"
+
+
+class SpecializedParserJobRunner:
+    def __init__(
+        self,
+        job_config: JobConfig,
+        processor_config: ProcessorConfig,
+        alloydb_config: AlloyDBConfig,
+        bigquery_config: BigQueryConfig,
+    ):
+        self.job_config = job_config
+        self.processor_config = processor_config
+        self.alloydb_config = alloydb_config
+        self.bigquery_config = bigquery_config
+
+        self.alloydb_connection_pool = self.create_connection_pool(alloydb_config)
+        self.storage_client = storage.Client(
+            client_info=ClientInfo(user_agent=USER_AGENT)
+        )
+        self.bq_client = bigquery.Client(client_info=ClientInfo(user_agent=USER_AGENT))
+
+    def run(self):
+        print("Verifying AlloyDB output table")
```

Comment:
change print to logging, and remove the ones that no longer relevant after the flow is verified

Reply:
Done - thanks